sklearn/tests/test_discriminant_analysis.py

```diff
@@ -331,7 +331,7 @@ def import_lda_module():
         return sklearn.lda
 
     lda = assert_warns(DeprecationWarning, import_lda_module)
-    assert lda.LDA is LinearDiscriminantAnalysis
+    assert lda.LDA.__class__ is LinearDiscriminantAnalysis.__class__
```

Comment:
This would be the correct test here I think -  `isinstance(LDA(), LinearDiscriminantAnalysis)`